utils/MyCurator.py

```diff
@@ -646,44 +697,34 @@ def get_roi_dicom_file(self, fw_object, study_uid, series_uid, sop_uid):
                 acq = acq.reload()
                 files.extend(acq.files)
             object_name = fw_object.label
-        
+
+        else:
+            log.warning(f"container type {container_type} is invalid for ROI importer")
+            return "INVALID CONTAINER TYPE"
+
         # Filter by file type, study UID and series UID:
-        files = [f for f in files if f.type == "dicom" and f.info.get("StudyInstanceUID") == study_uid and f.info.get("SeriesInstanceUID") == series_uid]
+        files = [f for f in files if f.type == "dicom" and f.info.get("StudyInstanceUID") == study_uid and f.info.get(
+            "SeriesInstanceUID") == series_uid]
         if len(files) == 0:
             log.warning(f"No dicom files found in session {object_name} with matching study/series UID "
                         f" Dicom Classifier must be run before ROI export.")
             return "NO MATCHES FOUND"
-        
+
         # If we found more than one dicom matching study/series, something is probably wrong...duplicate upload?
         if len(files) > 1:
             log.warning(f"more than 1 file found matching:\nStudyUID: {study_uid}\nSeriesUID: {series_uid}")
-        
+
         file = files[0]
-        acq = file.parent
+        if Path(file) == ".zip":
```

Comment:
two things,

* do you mean `Path(file).suffix == ".zip"`?
* I could see someone uploading a zip file with the extension not there, or an extension like `.zp` (which would probably break other things, but would be better to not break here)


You could do something like:

```python
try:
    match = self.match_zipped_dicom_member(acq, file, sop_uid)
except flywheel.rest.ApiException:
    # If it fails here, let it raise bc WTF
    match = self.match_unzipped_dicom(file, sop_uid)
```

Reply:
good catch on the suffix...and idk, the code above is a little too implicit vs explicit for me.  I'd rather be deliberate in our extension handling (this will cover > 80% 0f cases) and handle the outliers as they come rather than trying to predict what they may be.